model/generators/markets.py

```diff
@@ -250,10 +203,122 @@ def historical_returns(self, sample_size):
+        time_grid = TimeGrid(timesteps, timesteps)
+        if isinstance(process, StochasticProcessArray):
+            # time points are not really required for block step size but will if we change
+            # to other time delta
+            time_points = []
+            for index, _time in enumerate(time_grid):
+                time_points.append(time_grid[index])
+            steps = (len(time_points) - 1) * process.size()
+            sequence_generator = UniformRandomSequenceGenerator(
+                steps, UniformRandomGenerator())
+            gaussian_sequence_generator = GaussianRandomSequenceGenerator(
+                sequence_generator)
+            path_generator = GaussianMultiPathGenerator(
+                process, time_points, gaussian_sequence_generator, False)
+            paths = np.zeros(shape=(self.number_of_paths_per_asset,
+                                    process.size(), len(time_grid)))
+
+            for path_index in range(self.number_of_paths_per_asset):
+                multi_path_generator = path_generator.next().value()
+                for j in range(multi_path_generator.assetNumber()):
+                    path = multi_path_generator[j]
+                    paths[path_index, j, :] = np.array(
+                        [path[k] for k in range(len(path))])
+
+        log_returns = np.diff(np.log(paths[0]))
+
+        return log_returns
+
+    # def generate_uncorrelated_paths(self):
+    #         sequence_generator = UniformRandomSequenceGenerator(
+    #             len(time_grid), UniformRandomGenerator())
+    #         gaussian_sequence_generator = GaussianRandomSequenceGenerator(
+    #             sequence_generator)
+    #         maturity = time_grid[len(time_grid) - 1]
+    #         path_generator = GaussianPathGenerator(
+    #             process, maturity, len(time_grid), gaussian_sequence_generator, False)
+    #         paths = np.zeros(shape=(number_of_paths, len(time_grid)))
+    #         for path_index in range(number_of_paths):
+    #             path = path_generator.next().value()
+    #             paths[path_index, :] = np.array(
+    #                 [path[j] for j in range(len(time_grid))])
+
+    #     log_returns = np.diff(np.log(paths[0]))
+    #     # the current setup does only support simulation of processes with independent
+    #     # increments or processes without if the value is not changed in other sub-steps
+    #     increments = {}
+    #     for asset, path in list(zip(self.processes, log_returns)):
+    #         increments[asset] = path
+
+    #     return increments
+
+    # def switcher(self, params):
+    #     if (np.array(params['correlation']) -
+    #         np.diag(np.ones(len(params['correlation'])))
+    #             == np.zeros(np.array(params['correlation']).shape)).all():
+    #         process = self.process_container(params=params)
+    #     else:
+    #         process = None
+    #     return process
```

Comment:
Currently uncorrelated paths are produced with the multivariate path generator of QuantLib. Usually you would not to it like that, but simulate separate paths for performance. But I did not managed to get it to run and decided to leave it for later. We can move it into an issue?

Reply:
Sure, maybe move this code into an issue description and have it there for later. I want to avoid merging commented code to `main` moving forward. It builds up clutter.